### PR TITLE
checkmetrics percent modes

### PR DESCRIPTION
**Pull request by grahamwhaley**
Add the ability to:
 - show the results as percentages, not raw values
 - set the check ranges in the config file as percentages

### CODE REVIEW

**Comment by grahamwhaley** [comment]
Apologies in advance for sliding a `gofmt` fixup patch on the end of the series.... if you want me to split that and do a bunch of rebasing I can...

**Comment by egernst** [comment]
resulting diff looks fine, though perhaps some of it indeed should be squashed?  ie, adding in the toml also updates the metrics.go file?  Perhaps some of the calculation updates/changes should just be part of the initial commit (same for gofmt).

**Comment by grahamwhaley** [comment]
Looks like I hit a CI niggle as well:
```
cmd/checkmetrics/compare.go:161:12:warning: "acceptible" is a misspelling of "acceptable" (misspell)
```

so, let me break out some rebase foo and do a rework (and even try to run the static checks locally this time)

**Comment by grahamwhaley** [comment]
And more fun with CIs. I'm not convinced any of these are a problem with the PR.
The oddest is the Travis - its static checks seem unhappy with `math.Round()`, but they pass locally and it only works for me locally with `math.Round()` - I cannot drop to just `Round()` for instance. Maybe it is a go version thing? I am on 1.10.2 locally. Here is the output:

Travis go 1.8:
```
cmd/checkmetrics/compare_test.go:113:16:warning: Round not declared by package math (unused)
cmd/checkmetrics/compare_test.go:113:16:warning: unused struct field Round not declared by package math (structcheck)
cmd/checkmetrics/metric_test.go:43:16:warning: unused struct field Round not declared by package math (structcheck)
cmd/checkmetrics/metric_test.go:87:16:warning: unused struct field Round not declared by package math (structcheck)
cmd/checkmetrics/compare_test.go:113:16:warning: unused variable or constant Round not declared by package math (varcheck)
cmd/checkmetrics/metric_test.go:43:16:warning: unused variable or constant Round not declared by package math (varcheck)
cmd/checkmetrics/metric_test.go:87:16:warning: unused variable or constant Round not declared by package math (varcheck)
cmd/checkmetrics/compare_test.go:113:16:warning: Round not declared by package math (staticcheck)
```

And then for Travis go stable we got:
```
No output has been received in the last 10m0s, ...
```

And then, for Jenkins Ubuntu 16.04 we got a seemingly unrelated:
```
[etcd] Wrote Static Pod manifest for a local etcd instance to "/etc/kubernetes/manifests/etcd.yaml"
[init] Waiting for the kubelet to boot up the control plane as Static Pods from directory "/etc/kubernetes/manifests".
[init] This might take a minute or longer if the control plane images have to be pulled.
Build timed out (after 5 minutes). Marking the build as aborted.
Build was aborted
```

I'll kick a Jenkins rebuild and see if I can kick a Travis stable rebuild as well.

**Comment by grahamwhaley** [comment]
Hah, OK, `math.Round()` was [added in Go 1.10](https://github.com/golang/go/issues/4594#issuecomment-314186661) afaict, which is no doubt why the Travis go1.8 build is failing.
Our current minimum specified golang for this project is [1.8.3](https://github.com/kata-containers/runtime/blob/master/versions.yaml#L183). Hmm. OK, I'm going to presume I need to fix the code - if somebody would like to ack that is the case, I'll see if I can figure out how to do that as a build time thing (using `math.Round() if available, otherwise coding up a Round func).

**Comment by grahamwhaley** [comment]
OK, I added the relevant pair of test files and build constraints so we now provide our own Round() function if there is not one in the golang library. That should make Travis happier - and also mean this code actually works with our minimum go1.8.3 as specified.

**Comment by sboeuf** [comment]
@klynnrif PTAL

**Comment by egernst** [comment]
@intelkevinputnam - PTAL?

**Comment by intelkevinputnam** [comment]
Changes LGTM

File: cmd/checkmetrics/README.md [hunk 1]
```diff
@@ -82,6 +85,17 @@ the bounds set by the minval and maxval. That is:
...
 ```
 
+### Percentage presentation mode
+```
+--percentage        present results as percentage differences
+```
+
+> **Note:** Due to the way the internal comparisons are performed, if `minpercent` and
+> `maxpercent` are not equal, then the percentage range and midpoint value shown in the
+> results table will represent the *calculated* midpoint between the min and max, and
```
Comment (klynnrif) [inline cmd/checkmetrics/README.md]: results table represents the ...

File: cmd/checkmetrics/README.md [hunk 2]
```diff
@@ -82,6 +85,17 @@ the bounds set by the minval and maxval. That is:
...
+> not the supplied `midval`. This will make the results appear skewed, but the
```
Comment (klynnrif) [inline cmd/checkmetrics/README.md]: ... This makes the results appear skewed, but the

**Comment by jodh-intel** [comment]
Once @klynnrif's comments addressed...

lgtm

**Comment by grahamwhaley** [comment]
Updates applied @klynnrif - thx

**Comment by egernst** [comment]
...waiting for CI to turn green then will merge.

**Comment by egernst** [comment]
centos and fedora failed due to OBS it seems.  The required all passed, so going ahead and merging.

**Pull request merged**